gengokit/httptransport/templates/server.go

```diff
@@ -9,26 +9,34 @@ var ServerDecodeTemplate = `
 	// body. Primarily useful in a server.
 	func DecodeHTTP{{$binding.Label}}Request(_ context.Context, r *http.Request) (interface{}, error) {
 		var req pb.{{GoName $binding.Parent.RequestType}}
-		err := json.NewDecoder(r.Body).Decode(&req)
-		// err = io.EOF if r.Body was empty
-		if err != nil && err != io.EOF {
-			return nil, errors.Wrap(err, "decoding body of http request")
+		buf, err := ioutil.ReadAll(r.Body)
```

Comment:
This is good, cleaning up that old gross `json.NewDecoder().Decode()` stuff is a good move. :+1: